utils/src/main/java/software/amazon/awssdk/utils/NumericUtils.java

```diff
@@ -37,4 +38,63 @@ public static int saturatedCast(long value) {
         }
         return (int) value;
     }
+
+    /**
+     * Asserts that the given number is positive (non-negative and non-zero).
+     *
+     * @param num Number to validate
+     * @param fieldName Field name to display in exception message if not positive.
+     * @return Number if positive.
+     */
+    public static int assertIsPositive(int num, String fieldName) {
+        if (num <= 0) {
+            throw new IllegalArgumentException(String.format("%s must be positive", fieldName));
+        }
+        return num;
+    }
+
+    public static int assertIsPositive(int num, String fieldName, boolean zeroIsPositive) {
```

Comment:
Can we just have another method name assertIsNonNegative? Not a fan of flags to change method behavior.